pkg/kube/inject/inject.go

```diff
@@ -776,7 +776,7 @@ func IntoObject(sidecarTemplate string, valuesConfig string, revision string, me
 	if err != nil {
 		return nil, err
 	}
-	patchedObject, _, err := jsonSerializer.Decode(patched, nil, pod)
+	patchedObject, _, err := jsonSerializer.Decode(patched, nil, &corev1.Pod{})
```

Comment:
superimposing pod Specs is not going to work if we want to reorder containers. It will cause incorrect merges of podSpec config as the input pod does not have the istio-proxy nor does it share the same order as the patched pod